processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -58,18 +62,30 @@ public String getDimension()
     return this.dimension;
   }
 
+  @JsonProperty
+  public ExtractionFn getExtractionFn()
+  {
+    return extractionFn;
+  }
+
   public boolean eval(Row row)
   {
     List<String> dimRowValList = row.getDimension(this.dimension);
+    String extracted;
     if (dimRowValList == null || dimRowValList.isEmpty()) {
       return Strings.isNullOrEmpty(value);
     }
 
     for (String rowVal : dimRowValList) {
-      if (this.value != null && this.value.equals(rowVal)) {
+      if (this.getExtractionFn() != null) {
+        extracted = this.getExtractionFn().apply(rowVal);
```

Comment:
If I have the same extraction function applied in the dimension and here, will this cause the extraction to be applied twice?